File: csharp/grains/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n <= 0)
        {
            throw new ArgumentOutOfRangeException("The n cannot be less than or equal with zero.");
        }
        if (n > 64)
        {
            throw new ArgumentOutOfRangeException("The n cannot be greater than 64.");
        }
        return 1UL * (ulong)Math.Pow(2, n - 1);
    }

    public static ulong Total()
    {
        ulong sum = 0L;
        for (int i = 1; i <= 64; i++)
        {
            sum += Square(i);
        }
        return sum;
    }
}

File: cpp/grains/grains.cpp
#include "grains.h"
#include <cmath>

namespace grains
{

    unsigned long long square(int n)
    {
        if (n <= 0 || n > 64)
        {
            return 0;
        }

        return pow(2, n - 1);
    }

    unsigned long long total()
    {
        long long sum = 0;

        for (int i = 1; i <= 64; i++)
        {
            sum += square(i);
        }

        return sum;
    }

} // namespace grains
